src/plugins/orangehrmLDAPAuthenticationPlugin/Api/LDAPTestConnectionAPI.php

```diff
@@ -0,0 +1,358 @@
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\LDAP\Api;
+
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\ValidatorTrait;
+use OrangeHRM\LDAP\Api\Model\LDAPTestConnectionModel;
+use OrangeHRM\LDAP\Api\Traits\LDAPDataMapParamRuleCollection;
+use OrangeHRM\LDAP\Dto\LDAPSetting;
+use OrangeHRM\LDAP\Dto\LDAPUserLookupSetting;
+
+class LDAPTestConnectionAPI extends Endpoint implements CollectionEndpoint
+{
+    use ValidatorTrait;
+    use LDAPDataMapParamRuleCollection;
+
+    public const PARAMETER_ENABLED = 'enable';
+    public const PARAMETER_HOSTNAME = 'hostname';
+    public const PARAMETER_PORT = 'port';
+    public const PARAMETER_ENCRYPTION = 'encryption';
+    public const PARAMETER_LDAP_IMPLEMENTATION = 'ldapImplementation';
+
+    public const PARAMETER_BIND_ANONYMOUSLY = 'bindAnonymously';
+    public const PARAMETER_BIND_USER_DISTINGUISHED_NAME = 'bindUserDN';
+    public const PARAMETER_BIND_USER_PASSWORD = 'bindUserPassword';
+
+    public const PARAMETER_USER_LOOKUP_SETTINGS = 'userLookupSettings';
+    public const PARAMETER_BASE_DISTINGUISHED_NAME = 'baseDN';
+    public const PARAMETER_SEARCH_SCOPE = 'searchScope';
+    public const PARAMETER_USER_NAME_ATTRIBUTE = 'userNameAttribute';
+    public const PARAMETER_USER_UNIQUE_ID_ATTRIBUTE = 'userUniqueIdAttribute';
+    public const PARAMETER_USER_SEARCH_FILTER = 'userSearchFilter';
+
+    public const PARAMETER_DATA_MAPPING = 'dataMapping';
+    public const PARAMETER_FIRST_NAME = 'firstName';
+    public const PARAMETER_MIDDLE_NAME = 'middleName';
+    public const PARAMETER_LAST_NAME = 'lastName';
+    public const PARAMETER_USER_STATUS = 'userStatus';
+    public const PARAMETER_WORK_EMAIL = 'workEmail';
+    public const PARAMETER_EMPLOYEE_ID = 'employeeId';
+
+    public const PARAMETER_GROUP_OBJECT_CLASS = 'groupObjectClass';
+    public const PARAMETER_GROUP_OBJECT_FILTER = 'groupObjectFilter';
+    public const PARAMETER_GROUP_NAME_ATTRIBUTE = 'groupNameAttribute';
+    public const PARAMETER_GROUP_MEMBERS_ATTRIBUTE = 'groupMembersAttribute';
+    public const PARAMETER_GROUP_MEMBERSHIP_ATTRIBUTE = 'groupMembershipAttribute';
+    public const PARAMETER_SYNC_INTERVAL = 'syncInterval';
+
+    public const ENCRYPTION_NONE = 'none';
+    public const ENCRYPTION_TLS = 'tls';
+    public const ENCRYPTION_SSL = 'ssl';
+
+    public const LDAP_IMPLEMENTATION_OPEN_LDAP = 'OpenLDAP';
+    public const LDAP_IMPLEMENTATION_ACTIVE_DIRECTORY = 'ActiveDirectory';
```

Comment:
reuse all constants from `LDAPConfigAPI` and remove const from here, since we are accepting same request body

Reply:
Yeah we can do that, But thought of creating a base class and extend with both LDAPConfigAPI and LDAPTestConnectionAPI. Seems like we can eliminate lots of duplicated coded here when starting phase two test connection implementation.What do you think?

Reply:
done